e2e-tests/cypress/support/insurance/check-your-answers-eligibility-summary-list.js

```diff
@@ -0,0 +1,142 @@
+import { FIELD_IDS } from '../../../constants';
+import eligibilityCheckYourAnswersPage from '../../e2e/pages/insurance/check-your-answers/eligibility';
+import { FIELDS_ELIGIBILITY as FIELDS } from '../../../content-strings/fields/insurance/eligibility';
+import application from '../../fixtures/application';
+import { LINKS, DEFAULT } from '../../../content-strings';
+import getSummaryListField from './get-summary-list-field';
+
+const {
+  WANT_COVER_OVER_MAX_AMOUNT,
+  WANT_COVER_OVER_MAX_PERIOD,
+  OTHER_PARTIES_INVOLVED,
+  LETTER_OF_CREDIT,
+  PRE_CREDIT_PERIOD,
+  COMPANIES_HOUSE_NUMBER,
+  BUYER_COUNTRY,
+  HAS_MINIMUM_UK_GOODS_OR_SERVICES,
+  VALID_EXPORTER_LOCATION,
+} = FIELD_IDS.INSURANCE.ELIGIBILITY;
+
+const assertRow = (fieldId, expectedKey, expectedValue, expectedChangeLinkText) => {
+  const { summaryList } = eligibilityCheckYourAnswersPage;
+  const row = summaryList[fieldId];
+
+  cy.checkText(
+    row.key(),
+    expectedKey,
+  );
+
+  /**
+   * if value exists then checkText
+   * else check if dash
+   */
+  if (expectedValue) {
+    cy.checkText(
+      row.value(),
+      expectedValue,
+    );
+  } else {
+    cy.checkText(
+      row.value(),
+      DEFAULT.EMPTY,
+    );
+  }
+
+  /**
+   * if change link text, if there is an expected value, then should contain change
+   * else if no expected value, should say add
+   * if no expectedChangeLinkText, then should not have changeLink
+   */
+  if (expectedChangeLinkText) {
+    if (expectedValue) {
+      cy.checkText(
+        row.changeLink(),
+        `${LINKS.CHANGE} ${expectedChangeLinkText}`,
+      );
+    } else {
+      cy.checkText(
+        row.changeLink(),
+        `${LINKS.ADD} ${expectedChangeLinkText}`,
+      );
+    }
+  } else {
+    row.changeLink().should('not.exist');
+  }
```

Comment:
We have the same functions - check value/dash, check change link text currently in 2 places. Are we able to extract this into it's own function i.e one generic `assertRow`, passing summaryList to it?

Not sure if this would be easier to do now just for this new feature and then update the other lists, or do it all in one go, what do you think?

Reply:
can we have a ticket please